### ISSUE TITLE
'@see' jsdoc tag confuses bracket pair colorization

### ISSUE THREAD

**Issue by stephen-dirtcreative** [microsoft/vscode#150499]
Issue Type: <b>Bug</b>

The use of `{@see reference}` causes the closing `}` to be counted as belonging to the code base rather than part of the comment. If the `@` is removed (`{see reference}`) the colorization problem self-corrects, however, that is no longer valid JSDoc.

Expected results would be either that...
1. The opening AND closing bracket pair are colorized and recognized as a match, or...
2. Neither bracket is recognized as they are in a comment

A simple reproduction (on my system, the closing `}` of the `{@see hello}` is highlighted as an unexpected bracket in the function doc comment, and it "matches" the opening object brace in the object property doc comment.

```
function hello () {
  console.log('hello world!')
}

/**
 * Calls {@see hello} function. The previous close bracket in this comment
 * highlights as unexpected and unmatched.
 */
function callHello () {
  hello()
}

// In the object below, the closing brace of the @see reference matches the
// opening brace of the helloNamespace object, and therefore the actual closing
// brace at the end of the object is marked unmatched
const helloNamespace = {
  /** {@see hello} */
  hello: hello
}
```

While an 'unexpected' is not terribly problematic, if the commented JSDoc @see tag is inside of an object, for example, documenting a method set as a property, the bracket will then match the outer object opening brace and things become broken below that point.

In case they are useful, here are my related settings from User settings.json:
<details>
<summary>Selected User settings.json values</summary>

|Setting|Value|
|---|---|
|"editor.detectIndentation"|false|
|"editor.fontLigatures"|false|
|"editor.guides.bracketPairs"|true|
|"editor.guides.bracketPairsHorizontal"|"active"|
|"editor.guides.highlightActiveBracketPair"|true|
|"editor.guides.indentation"|false|
|"editor.guides.highlightActiveIndentation"|true|
|"editor.wordSeparators"|"/\\\\()\\\"':,.;<>~!@#$%^&*\|+=[]{}`?-"|
|"editor.suggestSelection"|"first"|
|"editor.quickSuggestions"|`{"other": "inline",  "comments": "off",  "strings": "off"}`|
|"javascript.inlayHints.parameterNames.enabled"|"all"|
|"javascript.inlayHints.variableTypes.enabled"|true|
|"javascript.inlayHints.propertyDeclarationTypes.enabled"|true|
|"javascript.inlayHints.parameterTypes.enabled"|true|
|"javascript.inlayHints.functionLikeReturnTypes.enabled"|true|
|"javascript.inlayHints.enumMemberValues.enabled"|true|
|"javascript.inlayHints.parameterNames.suppressWhenArgumentMatchesName"|true|
|"workbench.colorCustomizations"|`{"editorBracketHighlight.foreground1": "#b58900",  "editorBracketHighlight.foreground2": "#cb4b16",  "editorBracketHighlight.foreground3": "#d33682",  "editorBracketHighlight.foreground4": "#268bd2",  "editorBracketHighlight.foreground5": "#2aa198",  "editorBracketHighlight.foreground6": "#859900",  "editorBracketHighlight.unexpectedBracket.foreground": "#dc322f"}`|
</details>

VS Code version: Code 1.67.2 (c3511e6c69bb39013c4a4b7b9566ec1ca73fc4d5, 2022-05-17T18:20:57.384Z)
OS version: Darwin x64 21.4.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|16.00GB (0.02GB free)|
|Process Argv|--crash-reporter-id 12ce464d-f766-4f9e-9339-ef95105505c0|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (19)</summary>

Extension|Author (truncated)|Version
---|---|---
classdiagram-ts|Ale|0.1.65
tsco|alj|1.0.20
ng-template|Ang|13.3.4
scratchpad|awe|0.1.0
npm-intellisense|chr|1.4.1
vscode-eslint|dba|2.2.2
svg-sprites-viewer|Dev|0.0.2
gitlens|eam|12.0.7
vscode-npm-script|eg2|0.3.25
prettier-vscode|esb|9.5.0
todo-tree|Gru|0.0.215
vscode-guid|hea|1.4.20
plantuml|jeb|2.17.2
svg|joc|1.4.18
git-graph|mhu|1.30.0
remote-containers|ms-|0.234.0
live-server|ms-|0.2.12
vscode-stylelint|sty|1.2.2
vim|vsc|1.22.2

(1 theme extensions excluded)

</details>

**Comment by hediet** [comment]
The closing bracket has the scope
```
entity.name.type.instance.jsdoc comment.block.documentation.ts source.ts
```

However, the standard token type is `Other`, not `Comment`. This should be fixed.

**Comment by p4nu** [comment]
If it's okay I can have a look at this. 🙂

**Comment by p4nu** [comment]
Waiting for https://github.com/microsoft/vscode-dev-containers/pull/1483 to be merged so I don't have to setup a local development environment 🙂

**Comment by p4nu** [comment]
I have not yet resolved the problem with Codespaces.. 😕

**Comment by p4nu** [comment]
For what I've looked `{@see hello}` is not valid jsdoc. It's supposed to be `@see {@link hello}` according to [official JSDoc documentation](https://jsdoc.app/tags-see.html).

Correct me if I'm wrong here 🤔

**Comment by stephen-dirtcreative** [comment]
﻿You are correct. I am not sure what my brain was doing. I will have to check if @link has the same issue.

**Comment by p4nu** [comment]
It appears to be working with `@link`

<img width="325" alt="image" src="https://user-images.githubusercontent.com/28947061/172018631-870e00e6-e746-4602-a9ea-9e4e5734af6c.png">

**Comment by hediet** [comment]
Ah. Please close this issue if the JS doc was invalid indeed and all the other cases work.

**Comment by stephen-dirtcreative** [comment]
Yes, I was in error in my usage. Since proper use does not exhibit the problem, this issue can close. Very sorry for taking up your time, and thanks for your efforts!
